Sources/Diagnose/SourcekitdRequestCommand.swift

```diff
@@ -0,0 +1,72 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import ArgumentParser
+import Foundation
+import SourceKitD
+
+import struct TSCBasic.AbsolutePath
+
+public struct SourceKitdRequestCommand: AsyncParsableCommand {
+  public static var configuration = CommandConfiguration(
+    commandName: "run-sourcekitd-request",
+    abstract: "Run a sourcekitd request and print its result",
+    shouldDisplay: false
+  )
+
+  @Option(
+    name: .customLong("sourcekitd"),
+    help: "Path to sourcekitd.framework/sourcekitd"
+  )
+  var sourcekitdPath: String
+
+  @Option(
+    name: .customLong("request-file"),
+    help: "Path to a JSON sourcekitd request"
```

Comment:
Our requests are really yaml aren't they? Why do we keep calling it JSON everywhere?